File: chaos-ticker-backend/src/main/java/dev/chaos_ticker/engine/model/Reason.java
package dev.chaos_ticker.engine.model;

import java.util.List;
import java.util.Optional;

public record Reason(
  String text,
  Category category,
  int sharePriceImpact,
  List<RelevantSector> relevantSectors,
  Optional<Company> affectedCompany
) {}

File: chaos-ticker-frontend/src/types/Reason.ts
import { RelevantSectors } from "./RelevantSectors";
import { Company } from "./Company";

export interface Reason {
  text: string;
  category: string;
  magnitude: string;
  sharePriceImpact: number;
  newPrice: number;
  relevantSectors: RelevantSectors[];
  affectedCompany: Company
}